dev/com.ibm.ws.concurrent.persistent_fat/test-applications/schedtest/src/web/SchedulerFATServlet.java

```diff
@@ -2086,8 +2065,21 @@ public void testFindByName(PrintWriter out) throws Exception {
             throw new Exception("Should not be able to see tasks from outside of application. " + resultsByTaskId);
 
         // Wait for tasks C and D to complete their only execution.
-        pollForResult(statusC.getTaskId(), 1);
-        pollForResult(statusD.getTaskId(), 1);
+        for (long start = System.nanoTime(); System.nanoTime() - start < TIMEOUT_NS
+                && (statusC = scheduler.getStatus(statusC.getTaskId())) != null
+                && !(statusC.hasResult() && Integer.valueOf(1).equals(statusC.getResult())); )
+            Thread.sleep(POLL_INTERVAL);
+
+        if (!statusC.hasResult() || !Integer.valueOf(1).equals(statusC.getResult()))
```

Comment:
Should we first check if statusC == null before we check statusC.hasResults()?

Reply:
It isn't necessary because an erroneously null statusC will cause a NullPointerException and fail the test on this line, but I can add it if you prefer.